src/Network/Transport/Internal.hs

```diff
@@ -68,42 +73,66 @@ foreign import ccall unsafe "ntohs" ntohs :: CShort -> CShort
 #endif
 
 -- | Serialize 32-bit to network byte order
-encodeInt32 :: Enum a => a -> ByteString
-encodeInt32 i32 =
+encodeWord32 :: Word32 -> ByteString
+encodeWord32 i32 =
   BSI.unsafeCreate 4 $ \p ->
-    pokeByteOff p 0 (htonl . fromIntegral . fromEnum $ i32)
+    -- fromIntegral :: Word32 -> CInt
```

Comment:
Why this comment and `fromIntegral` is here?
`Network.Socket.htonl` is `Word32 -> Word32` so we can use `pokeByteOff p 0 (htonl w32)` here (btw, I think it worth renaming argument here)